rosbag2/src/rosbag2/writer.cpp

```diff
@@ -85,7 +105,13 @@ void Writer::remove_topic(const TopicMetadata & topic_with_type)
     throw std::runtime_error("Bag is not open. Call open() before removing.");
   }
 
-  storage_->remove_topic(topic_with_type);
+  auto it = topics_.find(topic_with_type.name);
+
+  if (it != topics_.end()) {
```

Comment:
Why are we removing topics which do not exist?

Reply:
This was a bug on my end. Fixing

Reply:
Actually hold on, `if (it != topics_.end()` is true when the topic exists.
If you meant the line `storage_->remove_topic(topic_with_type)` which happens inside the if-statement, then its because it needs to remove the topic from the internal storage. In the case of `sqlite3`, this involves updating a table.

Reply:
What I meant in this case, is that we should have:

```c++
if (it == topics_.end()) {
 throw std::runtime_error("tried to delete a non-existing topic");
} else {
  topics_.erase(it);
  storage_->remove_topic(topic_with_type);
}

Reply:
Oh yeah, I can do that.

Reply:
Done!